File: faktorial.dart
import 'dart:io';

int faktorial(int n) {
  int hasil = 1;
  for (int i = 1; i <= n; i++) {
    hasil *= i;
  }
  return hasil;
}

void main() {
  stdout.write("Masukkan angka: ");
  int n = int.parse(stdin.readLineSync()!);

  print("Faktorial dari $n adalah: ${faktorial(n)}");
}

File: faktorial.py
def factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        return n * factorial(n-1)

while True:
    try:
        num = int(input("Masukkan bilangan bulat >= 0: "))
        if num < 0:
            print("Bilangan harus >= 0")
            continue
        print(f"Faktorial dari {num} adalah {factorial(num)}")
        break
    except ValueError:
        print("Masukkan bilangan bulat yang valid")
